tests/testthat/test-read_resource.R

```diff
@@ -350,12 +350,74 @@ test_that("read_resource() understands encoding", {
   p <- example_package
   resource <- read_resource(p, "deployments")
 
-  # Create package with non-default missing values
+  # Create package with non-default encoding
   p_encoding <- p
   p_encoding$directory <- "."
   p_encoding$resources[[1]]$path <- "data/deployments_encoding.csv"
   p_encoding$resources[[1]]$encoding <- "windows-1252"
   expect_identical(read_resource(p_encoding, "deployments"), resource)
+
+  # Create package with unknown encoding
+  p_unknown <- p
+  p_unknown$resources[[1]]$encoding <- "utf-8-sig"
+  warnings <- capture_warnings(read_resource(p_unknown, "deployments"))
+  expect_identical(
+    warnings[1],
+    "Unknown encoding `utf-8-sig`. Reading file(s) with UTF-8 encoding."
+  )
+  expect_identical(
+    suppressWarnings(read_resource(p_unknown, "deployments")),
+    resource
+  )
+})
+
+test_that("read_resource() handles decimalChar/groupChar properties", {
+  expected_value <- 3000000.3
+  p <- suppressMessages(read_package("data/mark.json"))
+
+  # Default decimalChar/groupChar
+  resource <- read_resource(p, "mark_default")
+  expect_identical(resource$num, expected_value) # 3000000.30
+  expect_identical(resource$num_undefined, expected_value) # 3000000.30
+
+  # Non-default decimalChar, default groupChar (which should not conflict)
+  warnings <- capture_warnings(read_resource(p, "mark_decimal"))
+  expect_identical(
+    warnings[1],
+    paste(
+      "Some fields define a non-default `decimalChar`.",
+      "Parsing all number fields with `,` as decimal mark."
+    )
+  )
+
+  resource <- suppressWarnings(read_resource(p, "mark_decimal"))
+  expect_identical(resource$num, expected_value) # 3000000.30
+  expect_identical(resource$num_undefined, expected_value) # 3000000.30
+
+  # Non-default decimalChar/groupChar
+  warnings <- capture_warnings(read_resource(p, "mark_decimal_group"))
+  expect_true(length(warnings) == 3) # 2 warnings + 1 parsing failure last field
```

Comment:
From my personal experience, this kind of checks can be fragile against refactoring.
If you can find test examples where only one warning at a time is returned instead of 2 warning and 1 parsing failure, would make the unit-tests very robust.

Reply:
I agree, but this one is specifically designed to raise 2 warnings (and a parsing error). So, I'll leave it.